app/app.rb

```diff
@@ -0,0 +1,81 @@
+require 'sinatra/base'
+require_relative 'models/peep.rb'
+require_relative 'models/user.rb'
+require_relative '../database_connection_setup.rb'
+
+class Chitter < Sinatra::Base
+  enable :sessions
+  enable :method_override
+
+  get '/' do 
+    @sorted_peeps = Peep.all.sort_by { |peep| peep.posted_on }.reverse
+    erb(:index)
+  end
+
+  get '/registrations/new' do
+    erb(:'registrations/new')
+  end
+
+  post '/registrations' do
+    user = User.create(username: params[:username], email: params[:email], name: params[:name], password: params[:password])
+    if user
+      session[:user_id] = user.id
+      session[:signup_fail] = false
+      redirect('/')
+    else
+      session[:signup_fail] = true
+      redirect('/registrations/new')
+    end
+  end
+
+  get '/login' do
+    erb(:'sessions/new')
+  end
+
+  post '/sessions' do
```

Comment:
[This stack-overflow answer](https://stackoverflow.com/a/7252311) suggests the session routes should be:
- `post /session` 
- `get /session/new`
- `delete /session`



(since each end-user only ever have one session)